### DISCUSSION TITLE
Alias declaration for app-vite

### DISCUSSION THREAD

**Discussion by fprijate** [quasarframework/quasar#12963] [CLI - Electron mode] [answered]
For Quasar with **webpack (Electron mode)** I defined some aliases
in quasar.conf.js:

```
const path = require('path') ;
const { configure } = require('quasar/wrappers');

module.exports = configure(function (ctx) {
  return {
    electron: {
      chainWebpackMain (chain ) {
        chain.resolve.alias
        .set('@LIBB',path.resolve(__dirname, './AppLibrary/libBack'))
        .set('@LIBC',path.resolve(__dirname, './AppLibrary/libCommon'))
 
      }
  }
});
```

What is the right way to define them in Vite project.
For frontend i defined in 
quasar.config.js:

```
build: {
   
      extendViteConf(viteConf) {
        Object.assign(viteConf.resolve.alias, {
          "@LIBF": path.resolve(__dirname, "./AppLibrary/libFront"),
          "@LIBC": path.resolve(__dirname, "./AppLibrary/libCommon"),

        });
      },
```

How do I define aliases for Electron

**Comment by rstoenescu** [comment, marked as answer]
The Main and Preload threads are not passing through Vite. They are compiled by [Esbuild](https://esbuild.github.io/api). Unfortunately, Esbuild does not have a direct "alias" feature, but you could use "define" instead?

https://quasar.dev/quasar-cli-vite/developing-electron-apps/configuring-electron

```js
// quasar.config.js
electron: {
  extendElectronMainConf (cfg) {
    // do something with Esbuild config
    // for the Electron Main thread
  },

  extendElectronPreloadConf (cfg) {
    // do something with Esbuild config
    // for the Electron Preload thread
  }
}
```

<img width="324" alt="image" src="https://user-images.githubusercontent.com/4665641/160537587-d18e8cf3-3b06-400a-8ec6-d0ef42eea442.png">

**Reply by fprijate** [reply]
Thanks 

I have found  a solution. There is plugin [alias path](https://nx-plugins.netlify.app/derived/esbuild.html#alias-pathl)

My code.

**file: quasar.config.json:**

```
const { esbuildPluginAliasPath } = require("esbuild-plugin-alias-path");


module.exports = configure(function (/* ctx */) {
  return {

	electron: {
           extendElectronMainConf(esbuildConf) {
           const aliases = {
              "@SE/": path.resolve(__dirname, "./src-electron/x/"),
           };
           esbuildConf.plugins = [
             ...esbuildConf.plugins,
             esbuildPluginAliasPath({ alias: aliases }),
          ];
      }, 
```
 
**file:  electron-main.js:**

` import { bla } from "@SE/bla.js";`

**Reply by fprijate** [reply]
A difference with Vite aliases is that esbuild aliases for dirs must end with /.

**Reply by rstoenescu** [reply]
Thanks for writing this for the others that may end up here. That's what I want to see. Community collab 🚀
